File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.java
class Solution {
    public String processStr(String s) {
        StringBuilder ans=new StringBuilder();
        for(char x:s.toCharArray()){
            if(x=='*'){
                if(ans.length()>0) ans.deleteCharAt(ans.length()-1);
            } 
            else if(x=='#') ans.append(ans);
            else if(x=='%') ans.reverse();
            else ans.append(x);
        }
        return ans.toString();
    }
}

File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.cpp
class Solution {
public:
    string processStr(string s) {
        string ans="";
        for(char &x:s){
            if(x=='*'){
                if(!ans.empty()) ans.pop_back();
            } 
            else if(x=='#') ans+=ans;
            else if(x=='%') reverse(ans.begin(),ans.end());
            else ans+=x;
        }
        return ans;
    }
};